lib/http3/server.c

```diff
@@ -396,7 +396,9 @@ static void destroy_tunnel(struct st_h2o_http3_server_stream_t *stream)
     struct st_h2o_http3_server_conn_t *conn = get_conn(stream);
     if (stream->tunnel->datagram_flow_id != UINT64_MAX) {
         khiter_t iter = kh_get(stream, conn->datagram_flows, stream->tunnel->datagram_flow_id);
-        assert(iter != kh_end(conn->datagram_flows));
+        /* the tunnel wasn't established yet */
+        if (iter == kh_end(conn->datagram_flows))
+            return;
         kh_del(stream, conn->datagram_flows, iter);
```

Comment:
I wonder if we should avoid early return here. The rationale is that the function is expected to dispose all the properties, where `datagram_flow_id` is just one among them. Therefore, I tend to believe that a statement (or a block) that disposes each element should not do an early return, and that applies to the last property as well for the sake of consistency.